### ISSUE TITLE
Feature Request: Configurable HNSW Parameters (M, efConstruction, efSearch) in Typesense for Improved Semantic Search

### ISSUE THREAD

**Issue by AmanChourasia-TomTom** [typesense/typesense#1497]
## Issue Description:
In my usage of Typesense's semantic search capabilities, I've observed that the default settings for the HNSW parameters (M, efConstruction, efSearch) are not producing optimal results. These defaults seem to be limiting the search accuracy during semantic search.

## Enhancement needed:
I request to enable user-configurable HNSW parameters (M, efConstruction, efSearch) in Typesense. This enhancement would offer greater flexibility in fine-tuning the search functionality, potentially improving the accuracy and response time of semantic search results.

**Comment by DmitryKey** [comment]
Looking at the commit history -- this ticket has been addressed?

**Comment by jasonbosco** [comment]
Yup. This is now available in `0.26.0.rc54`:

```json
{
  "name": "vector_search",
  "fields": [
    {
      "name": "vec",
      "type": "float[]",
      "num_dim": 768,
      "hnsw_params": {
        "ef_construction": 200,
        "M": 16
      }
    }
  ]
}
```

**Comment by jasonbosco** [comment]
Released in v26
